### DISCUSSION TITLE
How to detect multiple keys down on KeyDown event?

### DISCUSSION THREAD

**Discussion by ShrutiJaiswal1494** [AvaloniaUI/Avalonia#5920] [Q&A] [answered]
I want to detect multiple keys down on  `KeyDown` event in `DataGrid`. I have a `MenuItem` with `InputGesture` as `Ctrl+O` but when I press `Ctrl+O` the below code will have `e.Key` as `LeftCtrl` and `KeyModifiers` as `Control`. How can I detect the combination?

```
 private void DataGrid_KeyDown(object sender, KeyEventArgs e)
 {
     if (e.Key == Key.O && (e.KeyModifiers == KeyModifiers.Control))
     {
          Application.Current.Clipboard.SetTextAsync("xyz");
      }
}
```

**Comment by maxkatz6** [comment, marked as answer]
Your code works fine:
```
private void DataGrid_KeyDown(object sender, KeyEventArgs e)
{
    Debug.WriteLine($"{e.Key} {e.KeyModifiers}");
    if (e.Key == Key.O && (e.KeyModifiers == KeyModifiers.Control))
    {
        Application.Current.Clipboard.SetTextAsync("xyz");
    }
}
```

Outputs expected result (Ctrl was pressed first, O was pressed after that with Ctrl kept down):
```
LeftCtrl Control
O Control
```

Condition also works and "xyz" is added to the clipboard.
